src/Sema.zig

```diff
@@ -25687,8 +25696,14 @@ fn explainWhyTypeIsNotExtern(
                 try mod.errNoteNonLazy(src_loc, msg, "slices have no guaranteed in-memory representation", .{});
             } else {
                 const pointee_ty = ty.childType(mod);
-                try mod.errNoteNonLazy(src_loc, msg, "pointer to comptime-only type '{}'", .{pointee_ty.fmt(sema.mod)});
-                try sema.explainWhyTypeIsComptime(msg, src_loc, pointee_ty);
+                if (!ty.isConstPtr(mod) and pointee_ty.zigTypeTag(mod) == .Fn) {
+                    try mod.errNoteNonLazy(src_loc, msg, "pointer to extern function must be 'const'", .{});
```

Comment:
This shouldn't add any notes about the pointee type since `validateExternType` doesn't check it either.

Reply:
I suggest doing this check in this pr

Reply:
Yes, sorry this review was a bit confusing, I changed my mind halfway through.

The `typeRequiresComptime` check should be in an `else if` branch to match `validateExternType`.

Reply:
+